modules/game_screenshot/game_screenshot.lua

```diff
@@ -0,0 +1,204 @@
+    {id = ScreenshotType.GIFT_OF_LIFE_TRIGGERED, label = "Gift of Life Triggered", enableDefault = true}
+}
+-- LuaFormatter on
+
+-- @ widget
+local evento = nil
+local optionPanel = nil
+-- @
+-- @ variables
+local autoScreenshotDirName = "auto_screenshots"
+local autoScreenshotDir = g_resources.getWriteDir() .. "/" .. autoScreenshotDirName
+
+-- @
+
+screenshotController = Controller:new()
+
+function screenshotController:onInit()
+
+end
+
+function screenshotController:onTerminate()
+    destroyOptionsModule()
+
+    if evento then
+        removeEvent(evento)
+        evento = nil
+    end
+end
+
+function screenshotController:onGameStart()
+
+    if g_game.getClientVersion() < 1310 then
+        return
+    end
+    optionPanel = g_ui.loadUI('game_screenshot')
+    modules.client_options.addTab('Screenshot', optionPanel, '/images/icons/icon_misc')
+
+    for _, temp in ipairs(AutoScreenshotEvents) do
+        local label = g_ui.createWidget("ScreenshotType", optionPanel.allCheckBox)
+        local settingKey = temp.label:gsub("%s+", "")
+        local setings = g_settings.getBoolean(settingKey)
+        label.text:setText(temp.label)
+        label.enabled:setChecked(setings)
+        label.enabled:setId(temp.id)
+        temp.currentBoolean = setings
+
+    end
+
+    if not g_resources.directoryExists(autoScreenshotDir) then
+        g_resources.makeDir(autoScreenshotDirName)
+    end
+
+    -- is g_game or LocalPlayer ?
+    screenshotController:registerEvents(LocalPlayer, {
+        onTakeScreenshot = onScreenShot
+    })
+end
+
+function screenshotController:onGameEnd()
+    if g_game.getClientVersion() >= 1310 then
+        destroyOptionsModule()
+    end
+    if evento then
+        removeEvent(evento)
+        evento = nil
+    end
+    for _, evento in ipairs(AutoScreenshotEvents) do
+        local labelSinEspacios = evento.label:gsub("%s+", "")
+        g_settings.set(labelSinEspacios, evento.currentBoolean)
+    end
+
+end
+
+function onUICheckBox(widget, checked)
+    if not widget then
+        return
+    end
+    local id = tonumber(widget:getId())
+    for _, temp in ipairs(AutoScreenshotEvents) do
+        if temp.id == id then
+            temp.currentBoolean = checked
+            break
+        end
+    end
+
+end
+
+-- LuaFormatter off
+function resetValues()
+    for _, evento in ipairs(AutoScreenshotEvents) do
+        local labelSinEspacios = evento.label:gsub("%s+", "")
+        g_settings.set(labelSinEspacios, evento.currentBoolean)
+
+    end
+
+    for _, i in pairs(optionPanel.allCheckBox:getChildren()) do
+        for _, j in pairs(i:getChildren()) do
+            if j:getStyle().__class == 'UICheckBox' then
+                local id = tonumber(j:getId())
+                if id then
+                    for _, evento in ipairs(AutoScreenshotEvents) do
+                        if evento.id == id then
+        --      hadouken                            
```

Comment:
remove this comment

Reply:
![image](https://github.com/mehah/otclient/assets/114332266/225701e0-1eb8-4553-b45a-4516fb06370e)